forward_command_controller/src/forward_command_controller.cpp

```diff
@@ -12,106 +12,126 @@
     return CallbackReturn::ERROR;
   }
 
-  if (auto rh_ptr = robot_hardware_.lock()) {
-    const auto registered_joints = rh_ptr->get_registered_joint_names();
-
-    // check all requested joints are present
-    for (const auto & joint_name : joint_names) {
-      if (std::find(
-          registered_joints.cbegin(), registered_joints.cend(),
-          joint_name) == registered_joints.cend())
-      {
-        RCLCPP_ERROR_STREAM(
-          rclcpp::get_logger(
-            logger_name_), "joint '" << joint_name << "' not registered");
-        return CallbackReturn::ERROR;
-      }
-    }
-
-    // get joint handles
-    for (const auto & joint_name : joint_names) {
-      hardware_interface::JointHandle joint_handle(joint_name, interface_name);
-      if (rh_ptr->get_joint_handle(joint_handle) ==
-        hardware_interface::hardware_interface_ret_t::ERROR)
-      {
-        // uppon error, clear any previously requested handles
-        joint_cmd_handles_.clear();
-
-        RCLCPP_ERROR_STREAM(
-          rclcpp::get_logger(
-            logger_name_), "could not get handle for joint '" << joint_name << "'");
-        return CallbackReturn::ERROR;
-      }
-      joint_cmd_handles_.emplace_back(std::move(joint_handle));
-    }
-  } else {
-    RCLCPP_ERROR_STREAM(
-      rclcpp::get_logger(
-        logger_name_), "could not lock pointer to robot_hardware");
+  if (interface_name_.empty()) {
+    RCLCPP_ERROR_STREAM(get_lifecycle_node()->get_logger(), "'interface_name' parameter was empty");
     return CallbackReturn::ERROR;
   }
 
   joints_command_subscriber_ = lifecycle_node_->create_subscription<CmdType>(
-    "commands", rclcpp::SystemDefaultsQoS(),
+    "~/commands", rclcpp::SystemDefaultsQoS(),
     [this](const CmdType::SharedPtr msg)
     {
       rt_command_ptr_.writeFromNonRT(msg);
     });
 
-  RCLCPP_INFO_STREAM(
-    rclcpp::get_logger(
-      logger_name_), "configure successful");
+  RCLCPP_INFO_STREAM(get_lifecycle_node()->get_logger(), "configure successful");
   return CallbackReturn::SUCCESS;
 }
 
+controller_interface::InterfaceConfiguration
+ForwardCommandController::command_interface_configuration() const
+{
+  controller_interface::InterfaceConfiguration command_interfaces_config;
+  command_interfaces_config.type = controller_interface::interface_configuration_type::INDIVIDUAL;
+
+  for (const auto & joint : joint_names_) {
+    command_interfaces_config.names.push_back(joint + "/" + interface_name_);
+  }
+
+  return command_interfaces_config;
+}
+
+controller_interface::InterfaceConfiguration
+ForwardCommandController::state_interface_configuration() const
+{
+  return controller_interface::InterfaceConfiguration{
+    controller_interface::interface_configuration_type::NONE};
+}
+
+// Fill ordered_interfaces with references to the matching interfaces
+// in the same order as in joint_names
+template<typename T>
+bool get_ordered_interfaces(
```

Comment:
We should use the implementation in ControllerInterface if https://github.com/ros-controls/ros2_controllers/pull/112/files#r532388849 goes forward.

Reply:
I opened issue #114  for this. We will proceed as it is for now and then make changes just after ros-controls/ros2_control#236 is merged. So we have a functioning version in master.

Reply:
Thanks @destogl , I agree